third-party/README.md

```diff
@@ -0,0 +1,10 @@
+# Third-party LLVM dependencies
```

Comment:
@Endilll @AaronBallman What do you think about this README for third-party dependencies?